internal/mithril-aggregator-discovery/src/http_config_discoverer.rs

```diff
@@ -0,0 +1,199 @@
+use std::{collections::HashMap, time::Duration};
+
+use anyhow::Context;
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+
+use mithril_common::{StdResult, entities::MithrilNetwork};
+
+use crate::{AggregatorDiscoverer, AggregatorEndpoint};
+
+const DEFAULT_REMOTE_NETWORKS_CONFIG_URL: &str =
+    "https://raw.githubusercontent.com/input-output-hk/mithril/main/networks.json";
+
+/// Representation of the networks configuration file.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct NetworksConfigMessage {
+    #[serde(flatten)]
+    pub networks: HashMap<String, NetworkEnvironmentMessage>,
+}
+
+/// Representation of a network environment in the networks configuration file.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct NetworkEnvironmentMessage {
+    #[serde(rename = "mithril-networks")]
+    pub mithril_networks: Vec<HashMap<String, MithrilNetworkMessage>>,
+}
+
+/// Representation of a Mithril network in the networks configuration file.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct MithrilNetworkMessage {
+    pub aggregators: Vec<AggregatorMessage>,
+}
+
+/// Representation of an aggregator in the networks configuration file.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct AggregatorMessage {
+    pub url: String,
+}
+
+/// An implementation of the [AggregatorDiscoverer] trait which discovers aggregators from remote networks configuration.
+///
+/// The reference file is the `networks.json` file hosted in the Mithril GitHub repository.
+pub struct HttpConfigAggregatorDiscoverer {
+    configuration_file_url: String,
+}
+
+impl HttpConfigAggregatorDiscoverer {
+    const HTTP_TIMEOUT: Duration = Duration::from_secs(10);
+
+    /// Creates a new `HttpConfigAggregatorDiscoverer` instance with the configuration file URL.
+    pub fn new(configuration_file_url: &str) -> Self {
+        Self {
+            configuration_file_url: configuration_file_url.to_string(),
+        }
+    }
+
+    /// Builds a reqwest HTTP client.
+    fn build_client(&self) -> StdResult<Client> {
+        Ok(Client::builder().timeout(Self::HTTP_TIMEOUT).build()?)
+    }
+}
+
+impl Default for HttpConfigAggregatorDiscoverer {
+    fn default() -> Self {
+        Self::new(DEFAULT_REMOTE_NETWORKS_CONFIG_URL)
+    }
+}
+
+#[async_trait::async_trait]
+impl AggregatorDiscoverer for HttpConfigAggregatorDiscoverer {
+    async fn get_available_aggregators(
+        &self,
+        network: MithrilNetwork,
+    ) -> StdResult<Box<dyn Iterator<Item = AggregatorEndpoint>>> {
+        let client = self.build_client()?;
+        let networks_configuration_response = client
```

Comment:
If possible moving the type to the `let` would make reading easier.
```suggestion
        let networks_configuration_response: NetworksConfigMessage = client
```